apps/examples/testcase/le_tc/network/tc_net_pkt.c

```diff
@@ -0,0 +1,160 @@
+#include <lwip/pbuf.h>
+
+#include "tc_internal.h"
+
+#define PBUF_TEST 6
+
+/**
+* @testcase             tc_net_pkt_layer_transport_p
+* @brief		check for the memory allocation for the pbuf by transport layer
+* @scenario
+* @apicovered           pbuf_alloc()
+* @precondition
+* @postcondition
+*/
+void tc_net_pkt_layer_transport_p(void) {
+
+	struct pbuf *p = NULL;
+	p = pbuf_alloc(PBUF_TRANSPORT, PBUF_TRANSPORT_HLEN, PBUF_RAM);
+	TC_ASSERT_NEQ_CLEANUP("pbuf", p, NULL, pbuf_free(p));
+	TC_SUCCESS_RESULT();
+
+	pbuf_free(p);
+}
+
+/**
+* @testcase             tc_net_pkt_layer_ip_p
+* @brief		check for the memory allocation for the pbuf by ip layer
+* @scenario
+* @apicovered           pbuf_alloc()
+* @precondition
+* @postcondition
+*/
+void tc_net_pkt_layer_ip_p(void) {
+
+	struct pbuf *p = NULL;
+	p = pbuf_alloc(PBUF_IP, PBUF_IP_HLEN, PBUF_RAM);
+	TC_ASSERT_NEQ_CLEANUP("pbuf", p, NULL, pbuf_free(p));
+	TC_SUCCESS_RESULT();
+
+	pbuf_free(p);
+}
+
+/**
+* @testcase             tc_net_pkt_layer_link_p
+* @brief		check for the memory allocation for the pbuf by link layer
+* @scenario
+* @apicovered           pbuf_alloc()
+* @precondition
+* @postcondition
+*/
+void tc_net_pkt_layer_link_p(void) {
+
+	struct pbuf *p = NULL;
+	p = pbuf_alloc(PBUF_LINK, PBUF_LINK_HLEN, PBUF_RAM);
+	TC_ASSERT_NEQ_CLEANUP("pbuf", p, NULL, pbuf_free(p));
+	TC_SUCCESS_RESULT();
+
+	pbuf_free(p);
+}
+
+/**
+* @testcase             tc_net_pkt_layer_raw_tx_p
+* @brief		check for the memory allocation for the pbuf by encapsulating link layer headers (e.g. 802.11)
+* @scenario
+* @apicovered           pbuf_alloc()
+* @precondition
+* @postcondition
+*/
+void tc_net_pkt_layer_raw_tx_p(void) {
+
+	struct pbuf *p = NULL;
+	p = pbuf_alloc(PBUF_RAW_TX, PBUF_LINK_ENCAPSULATION_HLEN, PBUF_RAM);
+	TC_ASSERT_NEQ_CLEANUP("pbuf", p, NULL, pbuf_free(p));
+	TC_SUCCESS_RESULT();
+
+	pbuf_free(p);
+}
+
+/**
+* @testcase             tc_net_pkt_layer_raw_p
+* @brief		check for the memory allocation for the pbuf by RX buffers
+* @scenario
+* @apicovered           pbuf_alloc()
+* @precondition
+* @postcondition
+*/
+void tc_net_pkt_layer_raw_p(void) {
+
+	struct pbuf *p = NULL;
+	p = pbuf_alloc(PBUF_RAW, 0, PBUF_RAM);
+	TC_ASSERT_NEQ_CLEANUP("pbuf", p, NULL, pbuf_free(p));
+	TC_SUCCESS_RESULT();
+
+	pbuf_free(p);
+}
+
+/**
+* @testcase             tc_net_pkt_layer_n
+* @brief		passed incorrect layer
+* @scenario
+* @apicovered           pbuf_alloc()
+* @precondition
+* @postcondition
+*/
+void tc_net_pkt_layer_n(void) {
+
+	struct pbuf *p = NULL;
+	p = pbuf_alloc(PBUF_TEST, 0, PBUF_RAM);
+	TC_ASSERT_EQ_CLEANUP("pbuf", p, NULL, pbuf_free(p));
+	TC_SUCCESS_RESULT();
+
+	pbuf_free(p);
+}
+
+/**
+* @testcase             tc_net_pkt_type_n
+* @brief		passed incorrect type. type parameter decides how and where the pbuf
+*                       should be allocated.
+* @scenario
+* @apicovered           pbuf_alloc()
+* @precondition
+* @postcondition
+*/
+void tc_net_pkt_type_n(void) {
```

Comment:
Except `net_pkt_main`, all of others are internal functions. Let's add `static`.

Reply:
done